dotnet/src/SemanticKernel/SkillDefinition/SKFunction.cs

```diff
@@ -245,11 +255,7 @@ public ISKFunction SetAIConfiguration(CompleteRequestSettings settings)
     /// <summary>
     /// Dispose of resources.
     /// </summary>
-    public void Dispose()
-    {
-        this.ReleaseUnmanagedResources();
-        GC.SuppressFinalize(this);
-    }
+    public void Dispose() => (this._aiService as IDisposable)?.Dispose();
```

Comment:
There could be unnecessary initialization of _aiService instance (Lazy<ITextCompletion>) if the being disposed function was created but never invoked.

Reply:
This apparently changed to be a `Lazy<>` between when I first put this up and now; this code is based on what was there pre-24 hours ago.  I'll update it.